src/fermilib/utils/_jellium_trotter_error.py

```diff
@@ -0,0 +1,461 @@
+            upper bound on the error (default) or to calculate the
+            norm of the error operator.
+
+    Returns:
+        A float upper bound on norm of error in the ground state energy.
+
+    Notes: follows Poulin et al.'s work in "The Trotter Step Size
+           Required for Accurate Quantum Simulation of Quantum
+           Chemistry". In particular, Equation 16 is used for a loose
+           upper bound, and the norm of Equation 9 is calculated for
+           a tighter bound using the error operator from
+           jellium_error_operator.
+
+           Possible extensions of this function would be to get the
+           expectation value of the error operator with the Hartree-Fock
+           state or CISD state, which can scalably bound the error in
+           the ground state but much more accurately than the triangle
+           inequality.
+    """
+    if tight:
+        # Return the 1-norm of the error operator (upper bound on error).
+        return numpy.sum(numpy.absolute(
+            list(jellium_error_operator(
+                terms, indices, is_hopping_operator).terms.values())))
+
+    zero = FermionOperator()
+    error = 0.0
+
+    for alpha in range(len(terms)):
+        term_a = terms[alpha]
+        coefficient_a, = term_a.terms.values()
+        if coefficient_a:
+            error_a = 0.0
+
+            for beta in range(alpha + 1, len(terms)):
+                term_b = terms[beta]
+                coefficient_b, = term_b.terms.values()
+                # If the terms don't commute, add the coefficient.
+                if not commutator(term_a, term_b).isclose(zero):
+                    error_a += abs(coefficient_b)
+
+            error += 4.0 * abs(coefficient_a) * error_a ** 2
+
+    return error
+
+
+def ordered_jellium_terms_with_info(
+        grid_length, dimension=3, wigner_seitz_radius=10.,
+        inverse_filling_fraction=2, spinless=True):
+    """Give terms from the jellium Hamiltonian in simulation algorithm order.
+
+    Roughly uses the simulation ordering. Pre-computes term info.
+
+    Args:
+        max_grid_length (int): The number of spatial orbitals per
+                               dimension.
+        dimension (int): The dimension of the system.
+        wigner_seitz_radius (float): The radius per particle in Bohr.
+        inverse_filling_fraction (int): The ratio of spin-orbitals to
+                                        particles.
+
+    Returns:
+        A 3-tuple of terms from the jellium Hamiltonian in order of
+        simulation, the indices they act on, and whether they are
+        hopping operators (both also in the same order).
+    """
+    zero = FermionOperator.zero()
+
+    n_qubits = grid_length ** dimension
+    if not spinless:
+        n_qubits *= 2
+
+    # Compute appropriate length scale.
+    n_particles = n_qubits // inverse_filling_fraction
+    length_scale = wigner_seitz_length_scale(
```

Comment:
I think this function should just take length_scale. That way it doesn't need to take two different parameters (the filling fraction and wigner_seitz)?  It can be up to the user of the function to compute the length scale.